### PR TITLE
Fix #2094: Make Scala Arrays instances of Serializable and Cloneable.

### LINKED ISSUE #2094: Scala Arrays are not instance of Cloneable and Serializable
As they should.

### PR DESCRIPTION
**Pull request by sjrd**

### CODE REVIEW

**Comment by sjrd** [comment]
Review by @nicolasstucki

File: test-suite/js/src/test/scala/org/scalajs/testsuite/compiler/ReflectionTest.scala [hunk 1]
```diff
@@ -122,6 +134,7 @@ object ReflectionTest extends JasmineTest {
       expect(classOf[String].cast(null)).toBeNull
       expect(classOf[String].cast("hello")).toEqual("hello")
       expect(classOf[Seq[_]].cast(List(1, 2)) == List(1, 2)).toBeTruthy
+      expect(() => classOf[Cloneable].cast(Array(3))).not.toThrow
```
Comment (nicolasstucki) [inline test-suite/js/src/test/scala/org/scalajs/testsuite/compiler/ReflectionTest.scala]: What about testing the cast to `Serializable`?

File: tools/scalajsenv.js [hunk 2]
```diff
@@ -871,7 +871,7 @@ initArray(
   // Runtime support
   this.constr = ArrayClass;
   this.parentData = ScalaJS.d.O;
-  this.ancestors = {O: 1};
+  this.ancestors = {O: 1, jl_Cloneable: 1, Ljava_io_Serializable: 1};
```
Comment (nicolasstucki) [inline tools/scalajsenv.js]: Maybe put them in the original order. Just for the sake oh homogeneouity.

**Comment by nicolasstucki** [comment]
That's all. Otherwise LGTM.

File: test-suite/js/src/test/scala/org/scalajs/testsuite/compiler/ReflectionTest.scala (hunk 1 above)
Comment (sjrd) [inline test-suite/js/src/test/scala/org/scalajs/testsuite/compiler/ReflectionTest.scala]: Indeed. Must have forgotten that one.

File: tools/scalajsenv.js (hunk 2 above)
Comment (sjrd) [inline tools/scalajsenv.js]: What would be the original order? There's not JavaDoc that we can consult, or anything. The best source of canon I found was run-time Java reflection:
```scala
scala> classOf[Array[Int]].getInterfaces
res0: Array[Class[_]] = Array(interface java.lang.Cloneable, interface java.io.Serializable)
```
which lists `Cloneable` before `Serializable`

Comment (nicolasstucki) [inline tools/scalajsenv.js]: I see.

**Comment by sjrd** [comment]
Updated.

**Comment by nicolasstucki** [comment]
LGTM

**Pull request merged**